From 8ea47983cbda4132767c70debf748604188453c4 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Tue, 15 Jan 2019 12:50:39 +0100
Subject: [PATCH] boards: tqma8mx: prepare config header for gpio port expander

this needs some tweaking in the config settings to separate between
SPL and U-Boot

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 configs/tqma8mq_mba8mx_mmc_defconfig |  3 +--
 include/configs/tqma8mx-mba8mx.h     |  5 +++++
 include/configs/tqma8mx.h            | 32 +++++++++++++++++++++++-----
 3 files changed, 33 insertions(+), 7 deletions(-)

diff --git a/configs/tqma8mq_mba8mx_mmc_defconfig b/configs/tqma8mq_mba8mx_mmc_defconfig
index 24e07a0ea7..b20c368ee5 100644
--- a/configs/tqma8mq_mba8mx_mmc_defconfig
+++ b/configs/tqma8mq_mba8mx_mmc_defconfig
@@ -15,8 +15,6 @@ CONFIG_CMD_GPIO=y
 CONFIG_CMD_I2C=y
 CONFIG_CMD_CACHE=y
 CONFIG_CMD_TIME=y
-CONFIG_CMD_PMIC=y
-CONFIG_CMD_REGULATOR=y
 CONFIG_CMD_UBI=y
 # CONFIG_DOS_PARTITION is not set
 CONFIG_OF_CONTROL=y
@@ -35,6 +33,7 @@ CONFIG_DM_REGULATOR=y
 CONFIG_DM_REGULATOR_PFUZE100=y
 CONFIG_DM_REGULATOR_FIXED=y
 CONFIG_DM_REGULATOR_GPIO=y
+CONFIG_SYS_I2C_MXC=y
 CONFIG_NXP_TMU=y
 CONFIG_USB=y
 CONFIG_DM_USB=y
diff --git a/include/configs/tqma8mx-mba8mx.h b/include/configs/tqma8mx-mba8mx.h
index d60cadd818..54958cfac5 100644
--- a/include/configs/tqma8mx-mba8mx.h
+++ b/include/configs/tqma8mx-mba8mx.h
@@ -7,4 +7,9 @@
 #if !defined(__TQMA8MX_MBA8MX_H)
 #define __TQMA8MX_MBA8MX_H
 
+#if !defined(CONFIG_SPL_BUILD)
+#define CONFIG_DM_GPIO
+#define CONFIG_DM_PCA953X
+#endif
+
 #endif /* __TQMA8MX_MBA8MX_H */
diff --git a/include/configs/tqma8mx.h b/include/configs/tqma8mx.h
index 99a5de3ec6..5814f4211a 100644
--- a/include/configs/tqma8mx.h
+++ b/include/configs/tqma8mx.h
@@ -50,11 +50,8 @@
 
 #define CONFIG_SPL_ABORT_ON_RAW_IMAGE /* For RAW image gives a error info not panic */
 
-#undef CONFIG_DM_MMC
-#undef CONFIG_DM_PMIC
-#undef CONFIG_DM_PMIC_PFUZE100
-
 #define CONFIG_SYS_I2C
+#define CONFIG_SYS_I2C_SPEED		100000
 #define CONFIG_SYS_I2C_MXC_I2C1		/* enable I2C bus 1 */
 #define CONFIG_SYS_I2C_MXC_I2C2		/* enable I2C bus 2 */
 #define CONFIG_SYS_I2C_MXC_I2C3		/* enable I2C bus 3 */
@@ -65,6 +62,31 @@
 #define CONFIG_POWER_I2C
 #define CONFIG_POWER_PFUZE100
 #define CONFIG_POWER_PFUZE100_I2C_ADDR 0x08
+
+#else
+
+/*
+ * TODO: DM stuff here to prevent SPL messing with missing stuff
+ * - pmic support
+ * - gpio expander support via dt
+ * - i2c eeprom support via dt
+ */
+#define CONFIG_CMD_PMIC
+#define CONFIG_CMD_REGULATOR
+
+#define CONFIG_DM_I2C
+#define CONFIG_DM_GPIO
+#define CONFIG_DM_MMC
+
+#define CONFIG_DM_PMIC
+#define CONFIG_DM_PMIC_PFUZE100
+#define CONFIG_PMIC_CHILDREN
+#define CONFIG_DM_REGULATOR
+#define CONFIG_DM_REGULATOR_PFUZE100
+#define CONFIG_DM_REGULATOR_FIXED
+#define CONFIG_DM_REGULATOR_GPIO
+#define CONFIG_I2C_EEPROM
+
 #endif
 
 #define CONFIG_REMAKE_ELF
@@ -263,7 +285,7 @@
 #define CONFIG_CMD_FUSE
 
 /* I2C Configs */
-#define CONFIG_SYS_I2C_SPEED		  100000
+/* #define CONFIG_SYS_I2C_SPEED		  100000 */
 
 /* USB configs */
 #ifndef CONFIG_SPL_BUILD
